app/views/users/show.html.erb

```diff
@@ -4,7 +4,7 @@
   </div>
 </div>
 
-<% if current_user == @user || !@user.private? || current_user.leaders.include?(@user) %>
+<% if policy(@user).show? %>
```

Comment:
Almost, but not quite. You need to add another policy method to ensure that the photo feed is displaying properly for both public and private users